### PR TITLE
Remove dependency on package:flutter_widgets.

### PR DESCRIPTION
**Pull request by kenzieschmoll**
We use two widgets from this package: `LinkedScrollControllerGroup` and `TaggedText`. Copying these widgets into our codebase eliminates the dependency on flutter_widgets, which is creating challenges with package rolls into Google.

### CODE REVIEW

**Comment by jamesderlin** [comment]
What's the long-term plan?  Is forking the code not going to create a maintenance burden in the future when the widget code is modified and eventually diverges?

File: packages/devtools_app/lib/src/flutter/flutter_widgets/linked_scroll_controller.dart [hunk 1]
```diff
@@ -0,0 +1,369 @@
+// Copyright 2020 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:flutter/rendering.dart';
+
+// This file is copied from package:flutter_widgets.
```
Comment (natebosch) [inline packages/devtools_app/lib/src/flutter/flutter_widgets/linked_scroll_controller.dart]: Will this remain relevant in a year? Should this detail be kept in the commit message exclusively?

**Comment by kenzieschmoll** [comment]
@jamesderlin I think development on package:flutter_widgets can continue as usual, and we can pull in changes to DevTools as needed. Since we only need two widgets from the package, I don't expect the maintenance burden will be that large. It would be nice if we could pull in the package directly, but that was causing some roadblocks. 

@grouma can you add some context as to why dropping the flutter_widgets dependency from DevTools is necessary?

**Comment by jamesderlin** [comment]
A few more things:
* FWIW, we have a long-term desire to break up `flutter_widgets` into separate packages.  That could make things slightly better (`TaggedText` doesn't get updated much).  OTOH `LinkedScrollContainer` has been more active, so it probably wouldn't help much.
* If your intention is that `flutter/devtools` will have its own private copy of these widgets and that it will periodically copy the `flutter_widgets`' version as needed, then, please:
  1. Add a highly visible disclaimer to the copied files that they should not be modified directly. (Some form of presubmit check would be even better...)
  2. Please note somewhere what `flutter_widgets` version or commit hash you forked from, either in the files, changelog, or commit message.

  Otherwise rectifying divergence will be much harder.

File: packages/devtools_app/lib/src/flutter/flutter_widgets/linked_scroll_controller.dart (hunk 1 above)
Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/flutter/flutter_widgets/linked_scroll_controller.dart]: added some context to the comments to state that they were "originally" copied from flutter_widgets and that the source may diverge over time.

**Comment by natebosch** [comment]
> we have a long-term desire to break up `flutter_widgets` into separate packages.  That could make things slightly better

As long as development continues to happen in `dart_src` that doesn't change the conversation. This package will still not be able to use it.

> If your intention is that `flutter/devtools` will have its own private copy of these widgets and that it will periodically copy the `flutter_widgets`' version as needed

I feel strongly that we should not make this a goal. Either we have a dependency (and we move devtools to be internal first in `dart_src`) or we don't (and we make this a clean fork which can be simplified to include only the features required by this package).

**Comment by jamesderlin** [comment]
> As long as development continues to happen in `dart_src` that doesn't change the conversation. This package will still not be able to use it.

Ah, I didn't understand before that this is motivated by rolls into Google.  Okay, that makes more sense.

Last September, I proposed making `flutter_widgets` use GitHub as its source-of-truth, but it was deemed unnecessary.  Perhaps we should reconsider that?

**Comment by jacob314** [comment]
If github was made the source of truth for flutter_widgets then we would happily unfork this code.

**Comment by jamesderlin** [comment]
[`linked_scroll_controller`](https://pub.dev/packages/linked_scroll_controller) now exists as a separate package.  The source-of-truth is still internal, but it internally has been moved out of `dart_src`, and my understanding is that the fork is no longer necessary.

**Pull request merged**